syntax = "proto3";

option java_multiple_files = true;
option objc_class_prefix = "HSW";

// service definition.
service DLServer {
  rpc proceed_streaming_task (stream DLStreamingTaskRequest) returns (stream DLStreamingTaskReply) {}
  rpc proceed_image_task (DLImageTaskRequest) returns (DLImageTaskReply) {}
  rpc proceed_video_task (DLVideoTaskRequest) returns (DLVideoTaskReply) {}
}

// The request message containing the image and deep learning task name
message DLStreamingTaskRequest {
  string task_name = 1;
  bytes image = 2;
}

// A response message containing the result
message DLStreamingTaskReply {
  bytes image = 1;
}

// The request message containing the image and deep learning task name
message DLImageTaskRequest {
  string task_name = 1;
  bytes image = 2;
  string bbox = 3;
}

// A response message containing the result
message DLImageTaskReply {
  bytes image = 1;
}

// The request message containing the deep learning task name and input video path
message DLVideoTaskRequest {
  string task_name = 1;
  string video_path = 2;
  string bbox = 3;    // Further improve -- send list bbox : dunno send list
}

// A response message containing the result
message DLVideoTaskReply {
  bool success = 1;
}
